cloudinit/url_helper.py

```diff
@@ -140,6 +140,15 @@ def read_ftps(url: str, timeout: float = 5.0, **kwargs: dict) -> "FtpResponse":
                     user=user,
                     passwd=url_parts.password or "",
                 )
+                LOG.debug("Creating a secure connection")
+                ftp_tls.prot_p()
+                LOG.debug("Reading file: %s", url_parts.path)
+                ftp_tls.retrbinary(
```

Comment:
The issue we have here in error handling is that "File not found" errors result in `ftplib.error_perm` permanent errors that we are casting as `URLError(cause="Attempted to connect to an insecure ftp server but used a scheme of ftps://"`  We actually need to look either at error code '550' perhaps or "No such file or directory" in str(e) and raise the appropriate code=NOT_FOUND instead of warnings about invalid ftp: scheme.

Consolidated diff suggestion for your thoughts below.

Reply:
Nice catch, thanks! This doesn't get raised to users from the testing code path that I'm fixing, but yes this is more correct for other cases.

I don't like using exception strings as a differentiator. What if we just use another try/except block to differentiate the exception message.